client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -121,10 +143,10 @@ public String getServiceAuthority() {
         @Override
         @Synchronized
         public void start(Listener listener) {
```

Comment:
Won't we have to make internal changes to gRPC if this is made a guava service? Anyway the class is thread safe now.